ampersand-dom-bindings.js

```diff
@@ -46,12 +51,22 @@ function switchHandler(binding, el, value) {
 }
 
 function getSelector(binding) {
-    if (typeof binding.selector === 'string') {
-        return binding.selector;
-    } else if (binding.hook) {
-        return '[data-hook~="' + binding.hook + '"]';
-    } else {
-        return '';
+    if(binding.firstMatchOnly !== undefined && binding.firstMatchOnly === true) {
+        if (typeof binding.selector === 'string') {
+            return {firstMatchOnly:binding.selector};
+        } else if (binding.hook) {
+            return {firstMatchOnly:'[data-hook~="' + binding.hook + '"]'};
+        } else {
+            return {firstMatchOnly:''};
+        }
+    }else{
+        if (typeof binding.selector === 'string') {
+          return binding.selector;
+        } else if (binding.hook) {
+          return '[data-hook~="' + binding.hook + '"]';
+        } else {
+          return '';
+        }
```

Comment:
Ah yes I see what you mean. You definitely thought about this way more than me when making this PR.  Sorry that my response was mostly notion and reaction, your detailed response was very appreciated.

I think that even though it requires changing more lines of code, adding a third parameter to `getMatches` that tells it whether or not to truncate results to the first result does seem like it would be easier to read/understand later, and less prone to unknown future edge cases.